frontend/public/components/operator-lifecycle-manager/create-operand.tsx

```diff
@@ -107,40 +147,74 @@ export const CreateOperandForm: React.FC<CreateOperandFormProps> = (props) => {
+      // NOTE: Can't use server-side error response due to missing info (https://github.com/kubernetes/kubernetes/issues/80718)
+      const fieldErrors = _.map(field.validation, (val, rule: Validations) => {
+        switch (rule) {
+          case Validations.minimum:
+            return formValues[field.path] >= val ? null : `Must be greater than ${val}.`;
+          case Validations.maximum:
+            return formValues[field.path] <= val ? null : `Must be less than ${val}.`;
+          case Validations.minLength:
+            return formValues[field.path].length >= val ? null : `Must be at least ${val} characters.`;
+          case Validations.maxLength:
+            return formValues[field.path].length <= val ? null : `Must be greater than ${val} characters.`;
+          case Validations.pattern:
+            return new RegExp(val as string).test(formValues[field.path]) ? null : `Does not match required pattern ${val}`;
+          default:
+            return null;
+        }
+      });
+      // Just use first error
+      return {...allErrors, [field.path]: fieldErrors.find(e => !_.isNil(e))};
+    }, {} as FieldErrors);
+    setFormErrors(errors);
 
-  const submit = (e) => {
-    e.preventDefault();
-
-    const specValues = _.omitBy(formValues, (val, path) => path.startsWith('metadata') || _.isNil(val));
-
-    const obj = {
-      apiVersion: apiVersionForModel(props.operandModel),
-      kind: props.operandModel.kind,
-      metadata: {
-        namespace: props.namespace,
-        name: formValues['metadata.name'],
-        labels: SelectorInput.objectify(formValues['metadata.labels']),
-      },
-      spec: _.reduce(specValues, (spec, value, path) => _.set(spec, path, value), {}),
-    } as K8sResourceKind;
-
-    k8sCreate(props.operandModel, obj)
-      .then(() => history.push(`/k8s/ns/${props.namespace}/clusterserviceversions/${props.clusterServiceVersion.metadata.name}/${referenceForModel(props.operandModel)}/${obj.metadata.name}`))
-      .catch((err: Status) => setError(err.message));
+    if (_.isEmpty(_.compact(_.values(errors)))) {
+      const specValues = fields.reduce((usedFormValues, field) => {
+        const formValue = _.get(usedFormValues, field.path);
+        if (_.isEqual(formValue, defaultValueFor(field))) {
+          return _.omit(formValues, field.path);
+        }
+        return usedFormValues;
+      }, _.omit(formValues, ['metadata.name', 'metadata.labels']));
+
+      const obj = {
+        apiVersion: apiVersionForModel(props.operandModel),
+        kind: props.operandModel.kind,
+        metadata: {
+          namespace: props.namespace,
+          name: formValues['metadata.name'],
+          labels: SelectorInput.objectify(formValues['metadata.labels']),
+          annotations: _.get(props.sample, 'metadata.annotations', {}),
+        },
+        spec: _.reduce(specValues, (spec, value, path) => _.set(spec, path, value), _.get(props.sample, 'spec', {})),
```

Comment:
Use the `spec` values from the included sample object to fill in fields that don't have form input components. Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1731886 and all related bugs.